@xen-orchestra/web-core/lib/components/ui/progress-bar/UiProgressBar.vue

```diff
@@ -26,11 +32,14 @@ const {
   value: _value,
   max = 100,
   showSteps,
+  displayMode = 'percent',
 } = defineProps<{
   legend: string
   value: number
   max?: number
+  percent?: number
   showSteps?: boolean
+  displayMode?: 'percent' | 'value'
 }>()
```

Comment:
According to our discussion on MM, I think `max` should be required when `displayMode` value is `'value'`, so here's my take on to make TS aware of this case:

```suggestion
interface Props {
  legend: string
  value: number
  showSteps?: boolean
}

interface PercentageProps {
  max?: number
  displayMode: 'percent'
}

interface ValueProps {
  max: number
  displayMode: 'value'
}

const { value: _value, max: _max, showSteps, displayMode } = defineProps<Props & (PercentageProps | ValueProps)>()
```
Remember to set a default value for `_max` when it is not provided.